src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -124,14 +130,36 @@ public static Tag parseTag(String tag) throws ParseException {
     }
 
     /**
-     * Parses a {@code String Diet} into an {@code Diet}.
+     * Parses a {@code String tag} into a {@code Tag}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code Diet} is invalid.
+     * @param dateTime Date and time of appointment.
+     * @param location Location of appointment.
+     * @throws ParseException if the given {@code tag} is invalid.
      */
-    public static Diet parseDiet(String diet) throws ParseException {
-        requireNonNull(diet);
-        String trimmedDiet = diet.trim();
-        return new Diet(trimmedDiet);
+    public static Appointment parseAppointment(String dateTime, String location) throws ParseException {
+        requireAllNonNull(dateTime, location);
+        try {
+            String formattedDateTime = formatDateTime(dateTime.trim());
+            String trimmedLocation = location.trim();
+            String appointmentDetails = formattedDateTime + " at "+ trimmedLocation;
+
+            if (!Appointment.isValidAppointment(appointmentDetails)) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        AppointmentCommand.MESSAGE_USAGE));
+            }
+            
+            return new Appointment(appointmentDetails);
+        } catch (Exception e) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AppointmentCommand.MESSAGE_USAGE), e);
+        }
```

Comment:
Hmm... I think there's no real need for the try-catch block here since it is merely catching and throwing the same ParseException in this case?

Reply:
Try catch was aimed to account for the formatDateTime() method. But I completely agreed this is code is weird and messy. I had cleaned it up.